### PR TITLE
docs: Update es role configuration

### LINKED ISSUE #4554: [BUG-python/deployment] elasticsearch role setup
**Describe the bug**
The [page](https://docs.argilla.io/en/latest/getting_started/installation/configurations/elasticsearch.html#configure-roles-and-users) that contains Elasticsearch role setup isnot correct. 
After creating separate role for argilla user and using it under `ARGILLA_ELASTICSEARCH` variable - I receive 403 error
If I use `superuser` role - all works.

Screenshot 2024-01-31 at 10 44 33

**Stacktrace and Code to create the bug**
```python
[01/31/24 09:53:10] WARNING  WARNING:opensearch:GET                  base.py:259
                             https://elasticsearch-es-http.eck:9200/            
                             [status:403 request:0.159s]                        
                    INFO     INFO:backoff:Backing off             _common.py:104
                             _setup_elasticsearch(...) for 11.3s                
                             (pydantic.errors.ConfigError: Your                 
                             Elasticsearch endpoint at                          
                             https://argilla:XXXX@elasticsearch-e               
                             s-http.eck:9200 is not available or                
                             not responding.                                    
                             Please make sure your Elasticsearch                
                             instance is launched and correctly                 
                             running and                                        
                             you have the necessary access                      
                             permissions.
```

**Expected behavior**
Running argilla server with connected elasticsearch engine by using custom created ES role with minimal required permissions.

**Environment:**
 - Argilla Version `1.22.0`
 - ElasticSearch Version `8.8.2`
 - Docker Image `argilla:v1.22.0`

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR reviews and update the role configuration for ES 

Closes https://github.com/argilla-io/argilla/issues/4554

**Type of change**

(Remember to title the PR according to the type of change)

- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes.)

- [ ] `sphinx-autobuild` (read [Developer Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation) for more details)

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: docs/_source/getting_started/installation/configurations/elasticsearch.md [hunk 1]
```diff
@@ -10,10 +10,7 @@ If you have an Elasticsearch instance and want to share resources with other app
 
 All you need to take into account is:
 
-
-* Argilla will create its ES indices with the following pattern `.argilla*`. It's recommended to create a new role (e.g., argilla) and provide it with all privileges for this index pattern.
-
-* Argilla creates an index template for these indices, so you may provide related template privileges to this ES role.
+* Argilla will create its ES indices with the following patterns: `rg.*` and `*ar.dataset*`. It's recommended to create a new role (e.g., argilla) and provide it with all privileges for these index patterns.
 
 Argilla uses the `ARGILLA_ELASTICSEARCH` environment variable to set the ES connection.
```
Comment (nataliaElv) [inline docs/_source/getting_started/installation/configurations/elasticsearch.md]:
```suggestion
- Argilla uses the `ARGILLA_ELASTICSEARCH` environment variable to set the ES connection.
```
I would add this as another bullet point in the "things to take into account". Another option is to get rid of the bullet point list, as we only have one item in the list right now.

**Review by nataliaElv — COMMENTED** [review]
LGTM, but I would suggest to ask @burtenshaw for another review, as he may be more familiar than me with the steps one needs to take to set up elasticsearch.

File: docs/_source/getting_started/installation/configurations/elasticsearch.md [hunk 2]
```diff
@@ -29,19 +26,8 @@ http(s)://user:passwd@elastichost
...
 
-![Argilla Role and permissions in ES](https://user-images.githubusercontent.com/2518789/142883104-f4f20cf0-34a0-47ff-8ee3-ab9f4644271c.png)
-
-
-## Change index analyzers
-
-By default, for indexing text fields, Argilla uses the `standard` analyzer for general search and the `whitespace`
-analyzer for more exact queries (required by certain rules in the weak supervision module). If those analyzers
-don't fit your use case, you can change them using the following environment variables:
-`ARGILLA_DEFAULT_ES_SEARCH_ANALYZER` and `ARGILLA_EXACT_ES_SEARCH_ANALYZER`.
```
Comment (burtenshaw) [inline docs/_source/getting_started/installation/configurations/elasticsearch.md:40]: Do we no longer support specifying analyzer or is this excess information?

Comment (frascuchon) [reply docs/_source/getting_started/installation/configurations/elasticsearch.md:40]: This is something that we do not support for feedback datasets. I would like to move the feedback dataset features supported as the default ones.

Comment (burtenshaw) [reply docs/_source/getting_started/installation/configurations/elasticsearch.md:40]: Make sense then.

**Review by burtenshaw — APPROVED** [review]

**Pull request merged**